lua/core/which-key.lua

```diff
@@ -118,6 +118,7 @@ M.config = function()
           "Workspace Diagnostics",
         },
         f = { "<cmd>silent FormatWrite<cr>", "Format" },
+        F = { "<cmd>lua vim.lsp.buf.formatting()<CR>", "Format (native-lsp)" },
```

Comment:
We don't need two bindings for formatting, instead 'f' should call a function which would dispatch to the preferred option

Reply:
I agree, but that would require some extra steps. What if I want to use `shfmt` but still let `clangd` deal with the formatting for me. We need some way to test if the client has formatting ability before having a fallback on an external tool.

Reply:
Ok, maybe this is out of the scope of this PR.
I'm not sure if I understand your example with `shfmt` and `clangd`, but the idea would indeed be that instead of calling `Format` we call an internal function that makes the dispatching (probably depending on a config value).